hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/BlockRecordManager.java

```diff
@@ -72,6 +72,14 @@ public interface BlockRecordManager extends BlockRecordInfo, AutoCloseable {
      */
     void startUserTransaction(@NonNull Instant consensusTime, @NonNull HederaState state);
 
+    /**
+     * "Advances the consensus clock" by updating the latest consensus timestamp that the node has handled. This should
+     * be called early on in the transaction handling process in order to avoid processing earlier transactions that have
+     * arrived out of order.
```

Comment:
You mean to avoid allocating same timestamp to multiple transactions right ?

Reply:
Good catch, done. Thanks!